iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1951,60 +1978,36 @@ public List<String> getPartitionNames(org.apache.hadoop.hive.ql.metadata.Table h
     Table icebergTable = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
     PartitionsTable partitionsTable = (PartitionsTable) MetadataTableUtils
             .createMetadataTableInstance(icebergTable, MetadataTableType.PARTITIONS);
-    Expression expression = generateExpressionFromPartitionSpec(icebergTable, partitionSpec);
-    Set<PartitionData> partitionList = Sets.newHashSet();
+    Expression expression = IcebergTableUtil.generateExpressionFromPartitionSpec(icebergTable, partitionSpec);
+    Set<Pair<PartitionData, Integer>> partitionList = Sets.newHashSet();
```

Comment:
can we use HashMap instead?

Reply:
Done; I moved this code into a separate method `getPartitionInfo` in `IcebergTableUtil` and made use of it here and also in `IcebergMajorQueryCompactor` because there was a lot of similarity in both places.

Reply:
👍